xknx/devices/binary_sensor.py

```diff
@@ -69,6 +69,11 @@ def _iter_remote_values(self):
         """Iterate the devices RemoteValue classes."""
         yield self.remote_value
 
+    @property
+    def unique_id(self):
+        """Return unique id for this device."""
+        return f"binary_sensor_{hash(self.remote_value.group_address_state)}"
```

Comment:
This is all still untested - not sure what `hash(GroupAddress)` will return and if it can be used.

Reply:
Binary sensor prefix is not needed. Unique IDs are scoped by platform

Reply:
Ah perfect! Thanks.

Reply:
Also it's just a string, so no need to hash it all? Or  du we have a tuple of its here?

Reply:
Well it would look more like an id with a hash - but yea doesn't really matter.

Reply:
I think it's better it is an understandable value.

Reply:
hash(GroupAddress) returns the raw GA value (int)
str returns "x/y/z" depending on used address Schema 
and __repr__ returns 'GroupAddress("x/y/z")' also depending on Schema.

Reply:
I guess str would be fine then, thanks for checking ;).

Reply:
I like the idea. But i see problem, that it is not guaranteed, that the unique id stays as is forever. (e.g. python one day might change the `__hash__()` of int. I would therfore use `md5` or any other guaranteed reproducible hash algorithm.

Am i right, it is not possible to add this on `device`-Level bc not all devices have a `group_address_state` ?

Reply:
We will just use the string of the GA, so no hash or md5 needed :). `Device` doesn't know anything about the `RemoteValues` of the real implemented devices.

Reply:
And, are we sure that there are installations out there, which use the same `group_address_state` for two devices? Should we guarantee that it is unique within one setup?

Reply:
And, last but not least. Assuming you have a light with a ga `1/2/3`. And you change this to a switch with the same address. What would be the expected behavior? The same or a different address?

Reply:
> And, are we sure that there are installations out there, which use the same group_address_state for two devices? Should we guarantee that it is unique within one setup?

There can be devices that use the same state GA but the unique_id is scoped to the platform - and I highly doubt that there are too binary sensors or anything with two equal state GAs

> And, last but not least. Assuming you have a light with a ga 1/2/3. And you change this to a switch with the same address. What would be the expected behavior? The same or a different address?

Since switch and light are operating on a different platform they will have a different address in HA.

Reply:
they will get same unique id scoped to platform. But since they are different platform, they will not retain any friendly name set from GUI, you have to re-set it. But switching would anyway have changed the prefix of the entity_id from switch.* to light.*

There is possibility that people may have used the same control address for a light switch with dimming and a light switch without dimming. This would then not be allowed. Also do we support light switches without on/off state, just a brightness level? It's somewhat tricky to decide which address is the base for unique id, the brightness level or the on/off command?